test/test_shape_ops.py

```diff
@@ -386,11 +386,11 @@ def test_flip(self, device):
         self.assertEqual(flip1_result, data.flip(1))
 
         # test empty tensor, should just return an empty tensor of the same shape
-        data = torch.tensor([])
+        data = torch.tensor([]).to(device)
```

Comment:
Thank you for fixing this. A better pattern would be:

```
data = torch.tensor((), device=device)
```

Then the tensor wouldn't have to be created and copied to the device.